cmd/common/common.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"math/rand"
+	"strings"
+
+	"github.com/ScottSallinen/lollipop/graph"
+)
+
+func ExtractGraphName(graphFilename string) (graphName string) {
+	gNameMainT := strings.Split(graphFilename, "/")
+	gNameMain := gNameMainT[len(gNameMainT)-1]
+	gNameMainTD := strings.Split(gNameMain, ".")
+	if len(gNameMainTD) > 1 {
+		return gNameMainTD[len(gNameMainTD)-2]
+	} else {
+		return gNameMainTD[0]
+	}
+}
+
+func ShuffleSC(sc []graph.StructureChange) {
+	for i := range sc {
+		j := rand.Intn(i + 1)
+		sc[i], sc[j] = sc[j], sc[i]
+	}
+}
+
+func InjectDeletesRetainFinalStructure(sc []graph.StructureChange, chance float64) []graph.StructureChange {
```

Comment:
Functions in pagerank_test.go use their own [StructureChange](https://github.com/ScottSallinen/lollipop/blob/186af2ec392566b3f8f860956b25f355cab0c8c7/cmd/lp-pagerank/pagerank_test.go#L46) type rather than [graph.StructureChange](https://github.com/ScottSallinen/lollipop/blob/186af2ec392566b3f8f860956b25f355cab0c8c7/graph/graph.go#L59). graph.StructureChange has an additional "weight" field. Does it matter if each StructureChange consumes more memory? 

I think it's okay for the tests we currently have since the graphs are very small. :)

Reply:
Yes, better to have PR use the proper graph version :) Old code probably.